Fix window resizing/moving issues and add ability to disable vibrancy

CvX opened the pull request:
Fixes #721 
Fixes #724 
Fixes #725 by reverting a faulty fix (my bad).

Also, improves the situation in #712 by providing a way to disable the vibrancy.

<img alt="vibrancy menu" width="400" src="https://user-images.githubusercontent.com/66961/51259065-b1869500-19ab-11e9-82fb-8feb0339eb86.png">

In a way, this PR acts as a stop-gap before a complete fix for vibrancy issues is done (though some people might like the vibrancy-less look so an option for that is a plus).

cc: @sindresorhus, @neegool

sindresorhus commented on config.js:
```diff
@@ -5,7 +5,7 @@ module.exports = new Store({
 		followSystemAppearance: false,
 		darkMode: false,
-		vibrancy: false,
+		vibrancy: 'none', // Possible values: 'none', 'sidebar', 'full'
```
I think we should default to `'sidebar'` to be as native-looking as possible by default.

You also need to add a migration from the boolean type, in this file. This is how I have handled migrations before: https://github.com/atomiclabs/hyperdex/blob/575056075278db372a724cde549743e0ba000bb2/app/config.js#L44-L59

sindresorhus replied:
> I think we should default to 'sidebar' to be as native-looking as possible by default.

Actually, let's wait to turn this on by default until the underlaying issue is fixed in Electron. Add a TODO comment about it however which links to the Electron issue, so we don't forget to change the default in the future.

CvX commented:
As requested, I've added the TODO comment regarding the default vibrancy setting and implemented the setting migration (`true` -> `'full'`, `false` -> `'sidebar'`).
